test_helpers/pool_helper.go

```diff
@@ -0,0 +1,244 @@
+package test_helpers
+
+import (
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/tarantool/go-tarantool"
+	"github.com/tarantool/go-tarantool/connection_pool"
+)
+
+type ListenOnInstanceArgs struct {
+	ConnPool *connection_pool.ConnectionPool
+	Mode connection_pool.Mode
+	ServersNumber int
+	ExpectedPorts map[string]bool
+}
+
+type CheckStatusesArgs struct {
+	ConnPool *connection_pool.ConnectionPool
+	Servers []string
+	Mode connection_pool.Mode
+	ExpectedPoolStatus bool
+	ExpectedStatuses map[string]bool
+}
+
+func compareTuples(expectedTpl []interface{}, actualTpl []interface{}) error {
+	if len(actualTpl) != len(expectedTpl) {
+		return fmt.Errorf("Unexpected body of Insert (tuple len)")
+	}
+
+	for i, field := range actualTpl {
+		if field != expectedTpl[i] {
+			return fmt.Errorf("Unexpected field, expected: %v actual: %v", expectedTpl[i], field)
+		}
+	}
+
+	return nil
+}
+
+func CheckPoolStatuses(args interface{}) error {
+	checkArgs, ok := args.(CheckStatusesArgs)
+	if !ok {
+		return fmt.Errorf("incorrect args")
+	}
+
+	connected, _ := checkArgs.ConnPool.ConnectedNow(checkArgs.Mode)
+	if connected != checkArgs.ExpectedPoolStatus {
+		return fmt.Errorf(
+			"incorrect connection pool status: expected status %t actual status %t",
+			checkArgs.ExpectedPoolStatus, connected)
+	}
+
+	poolInfo := checkArgs.ConnPool.GetPoolInfo()
+	for _, server := range checkArgs.Servers {
+		status := poolInfo[server] != nil && poolInfo[server].ConnectedNow
+		if checkArgs.ExpectedStatuses[server] != status {
+			return fmt.Errorf(
+				"incorrect conn status: addr %s expected status %t actual status %t",
+				server, checkArgs.ExpectedStatuses[server], status)
+		}
+	}
+
+	return nil
+}
+
+// ProcessListenOnInstance helper calls `return box.cfg.listen`
+// as many times as there are servers in the connection pool
+// with specified mode. For `PreferRO`/`RO` mode expected
```

Comment:
Well, but PreferRO is expected to receive master ports if there are no replicas

Reply:
Updated.